src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -0,0 +1,332 @@
+        events_entity = Entity("events", alias="events")
+        redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+        last_processed_id = int(redis_client.get(LAST_PROCESSED_REDIS_KEY) or 0)
+        for group_id_message_data_batch in chunked(
+            RangeQuerySetWrapper(
+                Group.objects.filter(
+                    project_id=project_id, id__gt=last_processed_id, type=ErrorGroupType.type_id
+                ).values_list("id", "message", "data"),
+                result_value_getter=lambda item: item[0],
+            ),
+            ITERATOR_CHUNK,
+        ):
+            group_id_message_batch = {
+                group_id: message
+                for (group_id, message, data) in group_id_message_data_batch
+                if not data.get("metadata").get("has_embeddings_record_v1")
+            }
+            group_id_batch = list(group_id_message_batch.keys())
+
+            query = Query(
+                match=events_entity,
+                select=[
+                    Column("group_id"),
+                    Function("max", [Column("event_id")], "event_id"),
+                ],
+                groupby=[Column("group_id")],
+                where=[
+                    Condition(Column("project_id"), Op.EQ, project_id),
+                    Condition(Column("group_id"), Op.IN, group_id_batch),
+                    Condition(
+                        Column("timestamp", entity=events_entity), Op.GTE, time - timedelta(days=90)
+                    ),
+                    Condition(Column("timestamp", entity=events_entity), Op.LT, time),
+                ],
+                orderby=[OrderBy(Column("group_id"), Direction.ASC)],
+            )
+
+            request = Request(
+                dataset=Dataset.Events.value,
+                app_id=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+                query=query,
+                tenant_ids={
+                    "referrer": Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+                    "cross_org_query": 1,
+                },
+            )
+
+            with metrics.timer(f"{BACKFILL_NAME}.bulk_snuba_queries", sample_rate=1.0):
+                result = bulk_snuba_queries(
+                    [request], referrer=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value
+                )
+
+            if result and result[0].get("data"):
+                rows: list[GroupEventRow] = result[0]["data"]
+                group_hashes = GroupHash.objects.filter(
+                    project_id=project_id, id__in=group_id_batch
+                )
+                group_hashes_dict = {
+                    group_hash.group_id: group_hash.hash for group_hash in group_hashes
+                }
+                data = lookup_group_data_stacktrace_bulk_with_fallback(
+                    project, rows, group_id_message_batch, group_hashes_dict
+                )
+                response = post_bulk_grouping_records(
```

Comment:
can we get timing for this function? can add the metrics.wraps decorator to it. like https://github.com/getsentry/sentry/blob/78b8cbeb9380058577e90c5d85f7c211ed2b1d21/src/sentry/replays/usecases/ingest/__init__.py#L178